File: backend/components/metiers/threed/utils/exporter/core/exporter.js
// exporter.js – Export CSV/JSON, anonymisation, RGPD, audit pour Threed (threed)
const fs = require('fs');

module.exports = {
  exportToJSON: (data, filepath) => {
    fs.writeFileSync(filepath, JSON.stringify(data, null, 2), 'utf-8');
    return filepath;
  },
  exportToCSV: (data, filepath) => {
    if (!Array.isArray(data) || data.length === 0) return filepath;
    const keys = Object.keys(data[0]);
    const csv = [keys.join(',')].concat(data.map(row => keys.map(k => row[k]).join(','))).join('\n');
    fs.writeFileSync(filepath, csv, 'utf-8');
    return filepath;
  },
  anonymize: (data) => {
    // Simule l'anonymisation RGPD
    return Array.isArray(data)
      ? data.map(row => ({ ...row, user: 'anonyme' }))
      : { ...data, user: 'anonyme' };
  }
};

File: backend/components/metiers/threed/utils/exporter/core/exporter.py
"""
exporter.py – Export CSV/JSON, anonymisation, RGPD, audit pour Threed (threed)
"""

import json
import csv


def export_to_json(data, filepath):
    with open(filepath, "w", encoding="utf-8") as f:
        json.dump(data, f, ensure_ascii=False, indent=2)
    return filepath


def export_to_csv(data, filepath):
    if not data:
        return filepath
    keys = data[0].keys()
    with open(filepath, "w", newline="", encoding="utf-8") as f:
        writer = csv.DictWriter(f, fieldnames=keys)
        writer.writeheader()
        writer.writerows(data)
    return filepath


def anonymize(data):
    # Simule l'anonymisation RGPD
    if isinstance(data, list):
        return [{**row, "user": "anonyme"} for row in data]
    return {**data, "user": "anonyme"}
